zstackctl/zstackctl/ctl.py

```
@@ -10308,17 +10311,55 @@ def build_catalina_opts(self, heap):
+                    warn('API update failed for GlobalConfig %s.%s, falling back to DB update' % (category, name))
+            else:
+                warn('cannot login to MN, falling back to DB update for GlobalConfig %s.%s' % (category, name))
+
+        try:
+            db_hostname, db_port, db_user, db_password = ctl.get_live_mysql_portal()
+        except CtlError as e:
+            raise CtlError('cannot update GlobalConfig %s.%s: %s' % (category, name, str(e)))
+
+        if db_password:
+            cmd = ShellCmd('''mysql -u %s -p%s --host %s --port %s zstack -e "update GlobalConfigVO set value='%s' where name='%s' and category='%s'"'''
+                           % (db_user, shell_quote(db_password), db_hostname, db_port, sval, name, category))
```

Ruff (0.15.9)

[error] Possible SQL injection vector through string-based query construction

(S608)